Rename RenderObject::absoluteRects() to RenderObject::boundingRects()

smfr opened the pull request:
#### 42d07b72904283f5ec6be263105b9e492f17bed6
<pre>
Rename RenderObject::absoluteRects() to RenderObject::boundingRects()
<a href="https://bugs.webkit.org/show_bug.cgi?id=259672">https://bugs.webkit.org/show_bug.cgi?id=259672</a>
rdar://113175046

Reviewed by Alan Baradlay.

`RenderObject::absoluteRects()` didn&apos;t return absolute rects; it returns whatever
the caller asked for via the second parameter. So rename to `boundingRects()`.
Also change the rects to be LayoutRects.

* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::elementPath const):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::hasNonEmptyBox):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::textFragmentIndicatorTimerFired):
p* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::boundingRects const):
(WebCore::RenderBlock::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::boundingRects const):
(WebCore::RenderBox::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::boundingRects const):
(WebCore::RenderInline::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderLineBreak.cpp:
(WebCore::RenderLineBreak::boundingRects const):
(WebCore::RenderLineBreak::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderLineBreak.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::absoluteBoundingBoxRect const):
(WebCore::RenderObject::absoluteTextRects):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::boundingRects const):
(WebCore::RenderObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::boundingRects const):
(WebCore::RenderText::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::boundingRects const):
(WebCore::RenderView::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderView.h:
* Source/WebCore/rendering/svg/LegacyRenderSVGModelObject.cpp:
(WebCore::LegacyRenderSVGModelObject::boundingRects const):
(WebCore::LegacyRenderSVGModelObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/LegacyRenderSVGModelObject.h:
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
(WebCore::RenderSVGBlock::boundingRects const):
(WebCore::RenderSVGBlock::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGBlock.h:
* Source/WebCore/rendering/svg/RenderSVGHiddenContainer.h:
* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::boundingRects const):
(WebCore::RenderSVGModelObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGModelObject.h:
* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::boundingRects const):
(WebCore::RenderSVGRoot::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGRoot.h:

Canonical link: <a href="https://commits.webkit.org/266494@main">https://commits.webkit.org/266494@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d26b7f22afe7cc6dee8a87c0c4db6746258e3690

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13952 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14267 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14602 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15690 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13244 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14036 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16776 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14350 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15913 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14122 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14716 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11819 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16394 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12002 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12580 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/19617 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13078 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12744 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15964 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13291 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11155 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12565 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3385 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16897 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13133 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay reviewed (approved):
1, I think the unionRect change deserves a dedicated patch (in case of revert)
2, This PR is a bit more than just a rename as it seems to be removing some of the integral snapping (didn't really follow to see if there was a "snap all the things" somewhere that accommodates for all the removed ones)

smfr committed:
Rename RenderObject::absoluteRects() to RenderObject::boundingRects()
https://bugs.webkit.org/show_bug.cgi?id=259672
rdar://113175046

Reviewed by Alan Baradlay.

`RenderObject::absoluteRects()` didn't return absolute rects; it returns whatever
the caller asked for via the second parameter. So rename to `boundingRects()`.
Also change the rects to be LayoutRects.

* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::elementPath const):
* Source/WebCore/html/HTMLAnchorElement.cpp:
(WebCore::hasNonEmptyBox):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::textFragmentIndicatorTimerFired):
p* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::boundingRects const):
(WebCore::RenderBlock::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderBlock.h:
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::boundingRects const):
(WebCore::RenderBox::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
* Source/WebCore/rendering/RenderInline.cpp:
(WebCore::RenderInline::boundingRects const):
(WebCore::RenderInline::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderInline.h:
* Source/WebCore/rendering/RenderLineBreak.cpp:
(WebCore::RenderLineBreak::boundingRects const):
(WebCore::RenderLineBreak::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderLineBreak.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::absoluteBoundingBoxRect const):
(WebCore::RenderObject::absoluteTextRects):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::boundingRects const):
(WebCore::RenderObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::boundingRects const):
(WebCore::RenderText::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::boundingRects const):
(WebCore::RenderView::absoluteRects const): Deleted.
* Source/WebCore/rendering/RenderView.h:
* Source/WebCore/rendering/svg/LegacyRenderSVGModelObject.cpp:
(WebCore::LegacyRenderSVGModelObject::boundingRects const):
(WebCore::LegacyRenderSVGModelObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/LegacyRenderSVGModelObject.h:
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
(WebCore::RenderSVGBlock::boundingRects const):
(WebCore::RenderSVGBlock::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGBlock.h:
* Source/WebCore/rendering/svg/RenderSVGHiddenContainer.h:
* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::boundingRects const):
(WebCore::RenderSVGModelObject::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGModelObject.h:
* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::boundingRects const):
(WebCore::RenderSVGRoot::absoluteRects const): Deleted.
* Source/WebCore/rendering/svg/RenderSVGRoot.h:

Canonical link: https://commits.webkit.org/266494@main
